### ISSUE TITLE
Apache mod PageSpeed Breaks Froxlor panel

### ISSUE THREAD

**Issue by mrpedersen38** [Froxlor/Froxlor#932]
I Just noticed this issue, and have seen others having the same issue, and resolving it by reinstalling.
Other users posted error logs and such.

I have checked all threads here on github, but cant find anything related.

When enabling apache mod "PageSpeed" Froxlor panel breaks.
UI items and more is broken, and SQL connection somehow breaks as well.
I experienced A ton of issues with my panel after activating PageSpeed.

This user had the same issue it seems, and not a "Browser" issue as first thought.
https://forum.froxlor.org/index.php?/topic/13053-probleme-mit-dashboard/

I Love this projekt ALOT, and i want to support the way i can.

Issue occurs in these browsers: Chrome, FireFox, Brave, IE, Edge, Safari.

I also got these errors in apache error log after enabling PageSpeed:

> froxlor[938]: SQLSTATE[HY000] [1045] Access denied for user 'root'@'127.0.0.1' (using password: YES)
> 
> froxlor[938]: --- DEBUG: #0 froxlor/lib/Froxlor/Database/Database.php(338): PDO->__construct('mysql:dbname=fr...', 'root', 'DB_ROOT_PWD...', Array) #1 /froxlor/lib/Froxlor/Database/Database.php(164): Froxlor\Database\Database::getDB() #2 /froxlor/admin_settings.php(28): Froxlor\Database\Database::needSqlData() #3 {main}
> 
> froxlor[1016]: SQLSTATE[HY000] [1045] Access denied for user 'root'@'127.0.0.1' (using password: YES)
> 
> froxlor[1016]: --- DEBUG: #0 froxlor/lib/Froxlor/Database/Database.php(338): PDO->__construct('mysql:dbname=fr...', 'root', 'DB_ROOT_PWD...', Array) #1 /froxlor/lib/Froxlor/Database/Database.php(164): Froxlor\Database\Database::getDB() #2 /froxlor/admin_settings.php(28): Froxlor\Database\Database::needSqlData() #3 {main}

**Comment by mrpedersen38** [comment]
Serversoftware: | Apache
-- | --
PHP-Version: | 7.3.27-1~deb10u1
MySQL server version: | 5.5.5-10.3.27-MariaDB-0+deb10u1
Webserver interface: | APACHE2HANDLER
Kernel: | 4.19.0-14-amd64 (x86_64)

Froxlor version:
Installed version: | 0.10.25-1 (DB: 202103030)
-- | --

**Comment by d00p** [comment]
Your mysql root password is wrong as the error message states...I doubt that has anything to do with mod_pagespeed at all

**Comment by mrpedersen38** [comment]
> Your mysql root password is wrong as the error message states...I doubt that has anything to do with mod_pagespeed at all

Hello d00p :)

when i deactivated PageSpeed the Access denied messages stopped, and everything works flawlessly.
When i saw those messages i thought that maybe the MySQL database were broken.
I tried resetting mysql root password, but it didn't help, and updating the userdata.inc.php file for froxlor.
The issue starts after clicking around the panel for a while, then it starts having problems logging in, and using domains page etc.

but, exactly when i deactivated PageSpeed, everything ran normally.
The server was installed:
root@web02:~# tune2fs -l /dev/sda1 | grep 'Filesystem created:'
Filesystem created:       Wed Jan 15 11:53:35 2020

So as you can see, froxlor has been installed for a while. (Froxlor was installed at the same time.)
Updates and everything else has been working flawlessly! Good work on that! :)

Before froxlor i did everything manually and by hand, so i know my way around services.

For now i am looking into excluding Froxlor in PageSpeed configs

**Comment by d00p** [comment]
Please provide steps to reproduce how you Integrated mod Pagespeed

**Comment by mrpedersen38** [comment]
Clean install, and just activate the mod:
wget https://dl-ssl.google.com/dl/linux/direct/mod-pagespeed-stable_current_amd64.deb
dpkg -i mod-pagespeed-stable_current_amd64.deb
a2enmod pagespeed

default config, only allowed my network to access pagespeed statistics.

All other web pages hosted on the server by froxlor panel works perfectly with the mod enabled.
Only the froxlor panel itself glitches.

Standard configured SQL, PHP and apache.
The SQL Error only occurs after a couple of hours i just noticed. (maybe cached credentials timeout some way?)
But other faults as UI Glitches and malfunctioning settings under "Domains" happens immediately. 
+ missing icons/graphics.

A quick solution is to just drop pagespeed mod, but i think that's just not right.

Basically just like this guide:
https://www.howtoforge.com/how-to-install-and-configure-mod-pagespeed-module-with-apache-on-ubuntu-1804/

**Comment by d00p** [comment]
Did you try the following by any chance?

> As of 1.13.35.1 PageSpeed is able to adapt its optimizations according to any Content Security Policies specified in the response headers. In this release you need to opt-in into this feature, future releases may turn it on by default.

```ModPagespeedHonorCsp on```

Or maybe even this one?

> When creating optimized .pagespeed. resources, PageSpeed generates headers that are correct for most users. However, some users require additional headers. For instance if you're using CORS and you want to have PageSpeed set Access-Control-Allow-Origin: http://www.example.com headers on the resources it creates, you can set:

```ModPagespeedAddResourceHeader "Access-Control-Allow-Origin" "http://www.example.com"```

**Comment by d00p** [comment]
no reply for over a month, close
